core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -588,9 +588,11 @@ object KafkaMetadataLog extends Logging {
     nodeId: Int
   ): KafkaMetadataLog = {
     val props = new Properties()
-    props.setProperty(TopicConfig.MAX_MESSAGE_BYTES_CONFIG, config.maxBatchSizeInBytes.toString)
-    props.setProperty(TopicConfig.SEGMENT_BYTES_CONFIG, config.logSegmentBytes.toString)
-    props.setProperty(TopicConfig.SEGMENT_MS_CONFIG, config.logSegmentMillis.toString)
```

Comment:
this is a critical bug - open https://issues.apache.org/jira/browse/KAFKA-19392 to fix it